### PR TITLE
Better error message for existing index

### PR DESCRIPTION
**Pull request by pontusmelke**
If you first created a composite index and then tried to create a node
key constraint on the same properties we failed when trying to create the
proper error message. Instead we got the error message:

```
Single property schema requires one property but had 2
```
which was caused by calling `getProperty` when creating the error message.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/schema/SchemaKernelException.java [hunk 1]
```diff
@@ -50,20 +53,29 @@ public SchemaKernelException( Status statusCode, String message )
...
 
-    protected static String messageWithLabelAndPropertyName( TokenNameLookup tokenNameLookup, String formatString,
+    static String messageWithLabelAndPropertyName( TokenNameLookup tokenNameLookup, String formatString,
             LabelSchemaDescriptor descriptor )
     {
+        int[] propertyIds = descriptor.getPropertyIds();
+
         if ( tokenNameLookup != null )
         {
+            String propertyString = propertyIds.length == 1 ?
```
Comment (craigtaverner) [inline community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/schema/SchemaKernelException.java:63]: I think this kind of logic should exist in one place. In an earlier version of this code we isolated the construction of a property description to the NodePropertyDescriptor which would internally know if it is single property or multi-property and format accordingly. Would that not be better here, so we don't repeat this logic in multiple places, and also don't have the conditional on the length of the property array?

File: community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/schema/SchemaKernelException.java [hunk 2]
```diff
@@ -50,20 +53,29 @@ public SchemaKernelException( Status statusCode, String message )
...
+                                            .collect( Collectors.joining( " and " ));
             return String.format( formatString,
-                    tokenNameLookup.labelGetName( descriptor.getLabelId() ),
-                    tokenNameLookup.propertyKeyGetName( descriptor.getPropertyId() ) );
+                    tokenNameLookup.labelGetName( descriptor.getLabelId() ), propertyString);
         }
         else
         {
+            String keyString = propertyIds.length == 1 ? "key[" + propertyIds[0] + "]" :
+                               "keys[" + Arrays.stream( propertyIds )
```
Comment (craigtaverner) [inline community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/schema/SchemaKernelException.java:74]: We also seem to be solving this in multiple places. See for example https://github.com/neo4j/neo4j/blob/f8512e67e1533d675e2f045fdb027391ed4b731b/community/kernel/src/main/java/org/neo4j/kernel/impl/util/dbstructure/DbStructureArgumentFormatter.java#L147

**Review by craigtaverner — APPROVED** [review]
Some suggestions that can be handled in later PRs

File: community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/schema/SchemaKernelException.java (hunk 1 above)
Comment (pontusmelke) [reply community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/schema/SchemaKernelException.java:63]: Yes probably, for the error message to have the correct pluralization we still need to check the length of the array though.

**Pull request merged**